### PR TITLE
Indexing

### PR DESCRIPTION
**Pull request by henrykironde**
add indexing to the engines and resolve the issues of the current indexing for file database systems

There was no indexing for sqlite, postgres and mysql. Current indexing for csv json and xml, was not resetting the index for the tables.

 set default encoding for test modules to avoid ASCII related issues.

 change the test dump directory to pytest's temporary directory

### CODE REVIEW

**Comment by henrykironde** [comment]
@ethanwhite, could the conflicts be the reason for the failures?

File: engines/sqlite.py [hunk 1]
```diff
@@ -10,13 +10,13 @@ class engine(Engine):
...
     datatypes = {
-        "auto": "INTEGER",
+        "auto": ("INTEGER", "AUTOINCREMENT"),
         "int": "INTEGER",
         "bigint": "INTEGER",
         "double": "REAL",
         "decimal": "REAL",
         "char": "TEXT",
-        "bool": "INTEGER",
+        "bool": "REAL",
```
Comment (ethanwhite) [inline engines/sqlite.py]: Seems like boolean should stays as "INTEGER" in SQLite:
https://www.sqlite.org/datatype3.html#section_2_1

Comment (henrykironde) [inline engines/sqlite.py]: I will check that

File: lib/engine.py [hunk 2]
```diff
@@ -365,7 +361,7 @@ def create_table_statement(self):
 
         create_stmt += ', '.join(column_strings)
         create_stmt += " );"
-
+        print (create_stmt)
```
Comment (henrykironde) [inline lib/engine.py]: I this will go

**Comment by ethanwhite** [comment]
Replaced by #546.

**Pull request closed without merging**